orangecontrib/bio/genialis/genapi.py

```diff
@@ -0,0 +1,190 @@
+from genesis import Genesis
+from genesis import GenData
+import json
+from Orange.data import *
+from .. import dicty
+import io
+import gzip
+
+DEFAULT_EMAIL = 'anonymous@genialis.com'
+DEFAULT_PASSWD = 'anonymous'
+DEFAULT_URL = 'https://dictyexpress.research.bcm.edu'
+CallBack = dicty.CallBack
+
+
+class Genapi(object):
+
+    """
+
+    Python module that leverages Genesis PyAPI (Python API for accsess to DictyExpress database).
+    It supports connection to the server and data retrieval functionalities.
+    """
+
+    def __init__(self, email=DEFAULT_EMAIL, password=DEFAULT_PASSWD, url=DEFAULT_URL, buffer=None):
+
+        self._gen = Genesis(email, password, url)
+        self.buffer = buffer
```

Comment:
We usually call this cache. I suggest not to use buffer/cache in the first version. Let's first identify the slow parts. We can plan how to add cache together. Maybe we can even avoid it altogether and use some 3rd party library like [requests-cache](https://pypi.python.org/pypi/requests-cache) or [slumbercache](https://pypi.python.org/pypi/slumbercache).